pkg/proxy/types.go

```diff
@@ -42,3 +43,33 @@ type ServicePortName struct {
 func (spn ServicePortName) String() string {
 	return fmt.Sprintf("%s:%s", spn.NamespacedName.String(), spn.Port)
 }
+
+// ServiceInfo is an interface which abstracts a service info.
+type ServiceInfo interface {
+	// ClusterIP returns service cluster IP.
+	ClusterIP() string
+	// Port returns service port.
+	Port() int
+	// Protocol returns service protocol.
+	Protocol() api.Protocol
+	// HealthCheckNodePort returns service health check node port.
```

Comment:
Is this "...if present" or is it always defined?  I forget

Reply:
> Is this "...if present" or is it always defined? I forget

This is "...if present". The comment message changed to 

```
// HealthCheckNodePort returns service health check node port if present.  If return 0, it means not present.
```

Is that ok?